code_samples/simple_language_plugin/src/main/java/org/intellij/sdk/language/SimpleStructureAwareNavbar.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+
+package org.intellij.sdk.language;
+
+import com.intellij.icons.AllIcons;
+import com.intellij.ide.navigationToolbar.StructureAwareNavBarModelExtension;
+import com.intellij.lang.Language;
+import com.intellij.psi.PsiComment;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.impl.source.tree.PsiCommentImpl;
+import com.intellij.psi.impl.source.tree.TreeElement;
+import org.intellij.sdk.language.psi.SimpleFile;
+import org.intellij.sdk.language.psi.SimpleProperty;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.Icon;
+
+public class SimpleStructureAwareNavbar extends StructureAwareNavBarModelExtension {
+    @NotNull
+    @Override
+    protected Language getLanguage() {
+        return SimpleLanguage.INSTANCE;
+    }
+
+    @Override
+    public @Nullable String getPresentableText(Object object) {
+        if (object instanceof SimpleFile) {
+            return ((SimpleFile) object).getName();
+        }
+        if (object instanceof SimpleProperty) {
+            return ((SimpleProperty) object).getName();
+        }
+
+        return null;
+    }
+
+    @Override
+    @Nullable
+    public Icon getIcon(Object object) {
+        if (object instanceof SimpleProperty) {
+            return AllIcons.Nodes.Property;
+        }
+
+        return null;
+    }
+
+    @Override
+    public @Nullable PsiElement adjustElement(@NotNull PsiElement psiElement) {
+        if (psiElement instanceof PsiComment) {
+            if (isCommentAttachedToProperty(psiElement)) {
```

Comment:
I would avoid this logic as it introduces IMHO redundant complexity and:
- would incorrectly display property if someone just commented out the previous property and not intended commenting the next property
- has a bug - when the caret is placed at the end of the comment line

This logic would be a nice feature for comments like KDoc, Javadoc, etc. where you are sure that it applies to the next element. Here, it is not the case.

Reply:
It was an idea to show the `adjustElement` method in action, but you're right that it doesn't really apply here, so I removed it.